bin/shutter

```diff
@@ -4149,12 +4152,7 @@ sub STARTUP {
 				if ($tray->signal_handler_is_connected($tray->{'hid2'})) {
 					$tray->signal_handler_block($tray->{'hid2'});
 				}
-			} elsif ($tray && $tray->isa('AppIndicator::Indicator')) {
```

Comment:
>> Why are you removing this?
Because this is legacy from Gtk2. In that module there really was the class `Gtk2::AppIndicator`, but in Gtk3, as I understood, this is just a stub.
` $ perl -w -Mstrict -le 'use Gtk3 "-init"; print Gtk3::AppIndicator->VERSION'
Use of uninitialized value in print at -e line 1.`

The condition `if (Gtk3::AppIndicator->VERSION > 0.12) { .. }` always will be false.
Moreover, if you put `$tray->set_active()` or ``$tray->set_passive()` directly, without checking the version of Gtk3::AppIndicator, you will get the error:
`if ($tray && $tray->isa('Gtk3::StatusIcon')) {
    ...
} elsif ($tray && $tray->isa('AppIndicator::Indicator')) {
    $tray->set_active();
}`

`Can't locate object method "set_active" via package "AppIndicator::Indicator" at bin/shutter line 4177.`

Reply:
`AppIndicator::Indicator` is still needed.

AppIndicator is required for Gnome, otherwise tray icon isn't visible - gnome doesn't support XEmbed legacy tray which is what Gtk3::StatusIcon is doing.

AppIndicator doesn't support all the features of Gtk3::StatusIcon, so Shutter tries both.

Reply:
See also https://ayatanaindicators.github.io/ - fork of AppIndicator used in some distros (Ubuntu)

Reply:
Ok, I suggest to close this PR. I'll fix this part and create a new one

Reply:
The `Gtk3::AppIndicator` part is Gtk2 legacy which I forgot to remove, that's different from `AppIndicator::Indicator`

Reply:
I returned removed code but with small additional. We check that VERSION is defined and only then compare versions

Reply:
Without it warnings in console will be emerged:
```
Use of uninitialized value in numeric gt (>) at bin/shutter line 4156.
 at bin/shutter line 4148.
	Shutter::App::fct_control_signals("block") called at bin/shutter line 807
	Shutter::App::STARTUP(Shutter::App=HASH(0x5581c1534ac0)) called at /home/aruzhnikov/perl5/lib/perl5/x86_64-linux-gnu-thread-multi/Glib/Object/Introspection.pm line 67
	Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x5581c1534ac0)) called at bin/shutter line 10944
...
Use of uninitialized value in numeric gt (>) at bin/shutter line 4180.
 at bin/shutter line 4172.
	Shutter::App::fct_control_signals("unblock") called at bin/shutter line 2720
	Shutter::App::STARTUP(Shutter::App=HASH(0x5581c1534ac0)) called at /home/aruzhnikov/perl5/lib/perl5/x86_64-linux-gnu-thread-multi/Glib/Object/Introspection.pm line 67
	Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x5581c1534ac0)) called at bin/shutter line 10944

Reply:
Erm, no, I mean, remove the condition (version check) altogether: 

``` perl
                        } elsif ($tray && $tray->isa('AppIndicator::Indicator')) {
					$tray->set_passive();
			}
```

Reply:
```perl
} elsif ($tray && $tray->isa('AppIndicator::Indicator')) {
		$tray->set_passive();
}

```
Can't locate object method "set_passive" via package "AppIndicator::Indicator" at bin/shutter line 4156.
 at bin/shutter line 4148.
	Shutter::App::fct_control_signals("block") called at bin/shutter line 807
	Shutter::App::STARTUP(Shutter::App=HASH(0x5605189942d0)) called at /home/aruzhnikov/perl5/lib/perl5/x86_64-linux-gnu-thread-multi/Glib/Object/Introspection.pm line 67
	Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x5605189942d0)) called at bin/shutter line 10940
Segmentation fault (core dumped)

Reply:
I think you found a bug here, that the AppIndicator variant of the tray icon didn't get disabled when it's supposed to.

Reply:
looks like it's something like this now (untested), `$tray->set_status('passive')` from what I can see in https://github.com/AyatanaIndicators/libayatana-appindicator/blob/master/src/app-indicator.h

Reply:
Yes, looks like this. With using method "set_status('active') / set_status('passive')" the program works correctly.
Have updated PR